Sources/eppo/dto/EppoValue.swift

```diff
@@ -152,21 +152,52 @@ public class EppoValue : Decodable, Equatable {
         return value;
     }
 
-    public func toHashedString() -> String {
-        // todo: this is a bit of a hack. we should probably just use a switch
-        // to handle the different types.
-        var str = ""
-        if let value = self.stringValue {
-            str = value
-        } else if let array = self.stringArrayValue {
-            str = array.joined(separator: " ,")
+    public func toEppoString() -> String {
+        switch self.type {
+        case .Boolean:
+            return try! self.getBoolValue() ? "true" : "false"
+        case .Numeric:
+            if let doubleVal = try? self.getDoubleValue() {
+                if floor(doubleVal) == doubleVal {
+                    // If the double value is an integer, return as integer string
+                    return String(format: "%.0f", doubleVal)
+                } else {
+                    // Otherwise, return the double as is
+                    return String(doubleVal)
+                }
+            } else {
+                // todo: should this throw?
+                return "null"
+            }
```

Comment:
When would the self.getDoubleValue fail?

Reply:
If the instance attribute doubleValue is null then it will throw; the type is a throwable double so I need to handle the failure case.

Practically this won't happen since we are in the code branch for doubles which matches the implementation above.

I can either return a sentinel value here or throw an exception, but that would make this function a throwable string.